### DISCUSSION TITLE
Auto-format of column width in pipe table

### DISCUSSION THREAD

**Discussion by rolandknapp** [quarto-dev/quarto-cli#4527] [Q&A] [answered]
I am drafting a Quarto document with a colleague, sharing versions and edits via Git/GitHub. Both of us are using the latest version of RStudio 2022.12.0+353 (OS: Ubuntu 22.04 for me, Mac OS for colleague). When passing document versions back and forth via Git push-pull, I've noticed that the column width specified in the header for a pipe table is auto-formatting between the two computers. For example, the screenshot below is a Git diff showing the table header as it was in my colleague's version (red background) and what it automatically changes to on my computer (green background). The reverse occurs when my colleague opens my version. I'm guessing that this is related to auto-formatting to fit a certain page width, but I can't find any Global Options that seem related to this. This auto-formatting wouldn't be an issue except that it automatically creates a new version of the document that needs to be committed and pushed. Any suggestions on how to avoid this behavior would be appreciated. 
![Screenshot from 2023-02-24 13-15-01](https://user-images.githubusercontent.com/38336080/221307576-25911883-34fc-4b48-816d-b001f9940d66.png)

**Comment by mcanouil** [comment, marked as answer]
The only thing I can think of that can change Markdown like that is activating/deactivating the visual editor.

I suspect one of you is using source while the other is using or at least switching to visual editor.

**Reply by rolandknapp** [reply]
I switch back and forth between the visual and source editor constantly, and you are correct that doing so results in the auto-format. Is there any way to avoid the resulting auto-format?

**Reply by mcanouil** [reply]
What you feel like a bug in RStudio is in fact that feature, namely the Visual Editor.
Thus if you do not want the formatting, do not use the visual editor.

Note that there is no reason to switch back and forth between source and visual, there are shortcuts, snippets, etc. To do all in both mode.

**Reply by rolandknapp** [reply]
I appreciate your point, but I suspect I'm not alone in preferring the visual versus source editor for different tasks (e.g., writing mostly R code versus writing mostly text). In addition, different users will have their own preferences for using the visual versus source editor, and ideally those preferences shouldn't cause version control issues in collaborative workflows.

**Reply by mcanouil** [reply]
I agree, but keep in mind that for the visual editor to work, some choices were made for a reason.

Since this is about RStudio, you can ask/report about this in RStudio repository, see <https://github.com/rstudio/rstudio>.
You'll probably get a better sense of why RStudio does that and if/how it can be changed in the future.
